src/Samples/Basic/Temporal.Samples.DesignProcess/Part1_5_AttributesAndInterfaces.cs

```diff
@@ -0,0 +1,287 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Hosting;
+
+using Temporal.Async;
+using Temporal.Common.DataModel;
+using Temporal.Worker.Hosting;
+using Temporal.Worker.Workflows;
+using Temporal.WorkflowClient;
+
+namespace Temporal.Sdk.BasicSamples
+{
+    /// <summary>
+    /// This sample goes together with <see cref="Part2_2_IfaceBasedClient" /.>
+    /// </summary>
+    public class Part1_5_AttributesAndInterfaces
+    {
+        // Entities:
+
+        public record MoneyAmount(int Dollars, int Cents) : IDataValue
+        {
+            public int TotalCents
+            {
+                get { return Dollars * 100 + Cents; }
+            }
+            
+            public static MoneyAmount operator +(MoneyAmount amountA, MoneyAmount amountB)
+            {
+                int sum = amountA.TotalCents + amountB.TotalCents;
+                return new MoneyAmount(sum / 100, sum % 100);                
+            }
+
+            public static bool operator >=(MoneyAmount amountA, MoneyAmount amountB)
+            {
+                return amountA.TotalCents >= amountB.TotalCents;
+            }
+
+            public static bool operator <=(MoneyAmount amountA, MoneyAmount amountB)
+            {
+                return amountA.TotalCents <= amountB.TotalCents;
+            }
+        }
+
+        public record OrderConfirmation(DateTime TimestampUtc, Guid OrderId) : IDataValue;
+
+        public record Product(string Name, MoneyAmount Price) : IDataValue;
+
+        public record Products(IReadOnlyCollection<Product> Collection) : IDataValue;
+
+        public record DeliveryInfo(string Address) : IDataValue;
+
+        public record User(string FirstName, string LastName, Guid UserId) : IDataValue
+        {
+            public string UserKey { get { return UserKey.ToString(); } }
+        }
+
+        // Workflow interfaces:
+
+        // These ifaces are used by the client (see 'Part2_2_IfaceBasedClient').
```

Comment:
Hrmm, I wonder if we can marry a dynamic proxy invoker approach and combined with an implicit context, you don't need this custom interface if you already have the workflow type, but instead can just make a `var myClient = new client.WorkflowClient<ShoppingCart>()` or something. Then you can `var cartRun = await myClient.Start()` and `var result = await cartRun.GetProductsAsync()` or something.

I think Java's approach shows a lot of promise over all of this manual typing (but I understand if there are limitations in .Net).